wcfsetup/install/files/lib/system/service/worker/Encryption.class.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace wcf\system\service\worker;
+
+use Jose\Component\Core\JWK;
+use Jose\Component\Core\Util\ECKey;
+use Jose\Component\KeyManagement\JWKFactory;
+use ParagonIE\ConstantTime\Base64;
+use ParagonIE\ConstantTime\Base64UrlSafe;
+use ParagonIE\ConstantTime\Binary;
+use wcf\data\service\worker\ServiceWorker;
+
+/**
+ * @author      Olaf Braun
+ * @copyright   2001-2024 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.1
+ */
+final class Encryption
+{
+    public const CURVE_ALGORITHM = 'P-256';
+    public const HASH_ALGORITHM = 'sha256';
+
+    /**
+     * Return the encrypted payload and necessary headers to send a push message to the push-service.
+     * {@link https://www.rfc-editor.org/rfc/rfc8291}
+     *
+     * @param ServiceWorker $serviceWorker
+     * @param string $payload
+     * @param array $headers to this array the encryption headers will be added
+     * @return string the encrypted payload
+     */
+    public static function encrypt(ServiceWorker $serviceWorker, string $payload, array &$headers): string
+    {
```

Comment:
As the `payload` is going to be encrypted, it should be marked as sensitive:
```suggestion
    public static function encrypt(
        ServiceWorker $serviceWorker,
        #[\SensitiveParameter]
        string $payload,
        array &$headers
    ): string {
```